rts/Sim/MoveTypes/GroundMoveType.cpp

```diff
@@ -813,25 +813,35 @@ void CGroundMoveType::SlowUpdate()
 			}
 
 			if (wantRepath) {
-				// When repaths are requested, they are pre-emptive and are made without
-				// confirmation that it is really necessary. Give the unit a chance to
-				// make progress: for example, when it got pushed against a building, but
-				// is otherwise moving on. Pathing is expensive so we really want to keep
-				// repathing to a minimum.
-				// Resolution distance checks kept to 1/10th of an Elmo to reduce the
-				// amount of time a unit can spend making insignificant progress, every
-				// SlowUpdate.
-				float curDist = math::floorf(currWayPoint.distance2D(owner->pos) * 10.f) / 10.f;
-				if (curDist < bestLastWaypointDist) {
-					bestLastWaypointDist = curDist;
-					wantRepathFrame = gs->frameNum;
+				// If the unit is being asked to walk into an exit-only zone, but finds itself outside of that, then
+				// force a repath: we know the path is no longer valid.
+				bool fallenOutOfExitOnly = (owner->moveDef->IsInExitOnly(currWayPoint) == true)
+				                        && (owner->moveDef->IsInExitOnly(owner->pos) == false);
```

Comment:
What if the destination is in a no-exit zone and we just want to get adjacent to it (i.e. trying to move into it is fine)?

Consider something like
```suggestion
				bool fallenOutOfExitOnly = (owner->moveDef->IsInExitOnly(currWayPoint) == true)
				                        && (owner->moveDef->IsInExitOnly(<FINAL PATH GOAL>) == false)
				                        && (owner->moveDef->IsInExitOnly(owner->pos) == false);
```

Reply:
We don't know if that we can get sufficiently close to the destination in that case. It is only possible to have a full and complete path to a destination in the exit-only zone if the entire path is fully in an exit-only zone. If the destination is in an exit-only zone and we can't reach that point (because we've falling out of the zone), then it is better for the pathing system to detect that on a path request and give the move system something it can realistically follow.